netket/graph/lattice.py

```diff
@@ -176,7 +229,10 @@ def __init__(
                 for atom_coord in atoms_coord
             ]
         )
-        if atoms_coord_fractional.min() < 0 or atoms_coord_fractional.max() >= 1:
+        if (
+            atoms_coord_fractional.min() < -cutoff_tol
+            or atoms_coord_fractional.max() > 1 + cutoff_tol
+        ):
```

Comment:
I guess you coudl do a hashing trick instead of this too if you wanted. Also, the way it is written now, it won't stop you from duplicating sites near the surface of the unit cell.

Reply:
Hmm, yeah that’s a good point. I’ll look into this tomorrow

Reply:
See below, I think it is good enough for now, and there is more fine-tuning about the whole hashing idea that needs to be done, and will probably become a small module in itself. We can come back to plugging it in here once that's done. Anyway, it's also the user's responsibility not to supply stupid things!